src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -267,6 +276,21 @@ private void checkRequiredParameterPresent(Map<String, Object> parameterMap, Lis
         }
     }
 
+    private void movePrivateKey(String rootPath, String privateKeyPath) {
+        Path permPrivateKeyPath = Paths.get(rootPath, PRIVATE_KEY_PATH_RELATIVE_TO_ROOT);
+        try {
+            if (Files.notExists(permPrivateKeyPath)) {
+                Files.createFile(permPrivateKeyPath);
+            }
+            Files.move(Paths.get(privateKeyPath), permPrivateKeyPath, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
I don't think you should use move as that will, well, move the file. IMO, a copy is appropriate.

Reply:
Right, with move, the file at the previous path won't exist but with copy, it will. Why do we need to preserve the previous the file at the path?

Reply:
I think that would be best under the circumstances.